tendrl/node_agent/discovery/sds/plugins/discover_gluster_sds_details.py

```diff
@@ -48,23 +50,46 @@ def discover_storage_system(self):
         ret_val['detected_cluster_name'] = "gluster-%s" % cluster_id
         ret_val['peers'] = gfs_peers
 
-        cmd = subprocess.Popen(
-            "gluster --version",
-            shell=True,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE
-        )
-        out, err = cmd.communicate()
-        if err and 'command not found' in err:
-            logger.log(
-                "debug",
-                NS.publisher_id,
-                {"message": "gluster not installed on host"}
+        # Check if the file /etc/tendrl/redhat-gluster-storage.json
+        # if the file exists, read the version details from this
+        if os.path.exists('/etc/tendrl/redhat-gluster-storage.json'):
```

Comment:
which rpm is going to provide this file? If it is not a Tendrl rpm then it should not be inside `/etc/tendrl`